tests/test_stability_funs.py

```diff
@@ -349,3 +351,402 @@ def test_magwell_print(capsys):
+            "grad(alpha)",
+            "grad(psi)",
+            "B",
+            "grad(|B|)",
+            "kappa",
+            "iota",
+            "lambda_t",
+            "lambda_z",
+            "lambda_tt",
+            "lambda_zz",
+            "lambda_tz",
+            "g^aa",
+            "g^ra",
+            "g^rr",
+            "cvdrift",
+            "cvdrift0",
+            "|B|",
+            "B^zeta",
+        ]
+
+        grid = Grid.create_meshgrid(
+            [rho, alpha, zeta],
+            coordinates="raz",
+            period=(np.inf, 2 * np.pi, np.inf),
+        )
+
+        data = eq.compute(data_keys, grid=grid)
+
+        psi_s = data_eq["psi"][-1]
+        sign_psi = psi_s / np.abs(psi_s)
+        sign_iota = iotas / np.abs(iotas)
+        # normalizations
+        Lref = data_eq["a"]
+        Bref = 2 * np.abs(psi) / Lref**2
+
+        modB = data["|B|"]
+        x = Lref * np.sqrt(psi)
+        shat = -x / iotas * shears / Lref
+
+        psi_r = data["psi_r"]
+
+        grad_psi = data["grad(psi)"]
+        grad_psi_sq = data["|grad(psi)|^2"]
+        grad_alpha = data["grad(alpha)"]
+
+        g_sup_rr = data["g^rr"]
+        g_sup_ra = data["g^ra"]
+        g_sup_aa = data["g^aa"]
+
+        modB = data["|B|"]
+
+        B_sup_zeta = data["B^zeta"]
+
+        gds2 = np.array(dot(grad_alpha, grad_alpha)) * Lref**2 * psi
+        gds2_alt = g_sup_aa * Lref**2 * psi
+
+        gds21 = -sign_iota * np.array(dot(grad_psi, grad_alpha)) * shat / Bref
+        gds21_alt = -sign_iota * g_sup_ra * shat / Bref * (psi_r)
+
+        gds22 = grad_psi_sq * (1 / psi) * (shat / (Lref * Bref)) ** 2
+        gds22_alt = g_sup_rr * (psi_r) ** 2 * (1 / psi) * (shat / (Lref * Bref)) ** 2
+
+        gbdrift = np.array(dot(cross(data["B"], data["grad(|B|)"]), grad_alpha))
+        gbdrift *= -sign_psi * 2 * Bref * Lref**2 / modB**3 * np.sqrt(psi)
+        gbdrift_alt = -sign_psi * data["gbdrift"] * 2 * Bref * Lref**2 * np.sqrt(psi)
+
+        cvdrift = (
+            -sign_psi
+            * 2
+            * Bref
+            * Lref**2
+            * np.sqrt(psi)
+            * dot(cross(data["B"], data["kappa"]), grad_alpha)
+            / modB**2
+        )
+        cvdrift_alt = -sign_psi * data["cvdrift"] * 2 * Bref * Lref**2 * np.sqrt(psi)
+
+        np.testing.assert_allclose(gds2, gds2_alt)
+        np.testing.assert_allclose(gds22, gds22_alt)
+        np.testing.assert_allclose(gds21, gds21_alt)
+        np.testing.assert_allclose(gbdrift, gbdrift_alt)
+        np.testing.assert_allclose(cvdrift, cvdrift_alt, atol=1e-2)
+
+        sqrt_g_PEST = data["sqrt(g)_PEST"]
+        np.testing.assert_allclose(sqrt_g_PEST, 1 / (B_sup_zeta / psi_r))
+
+
+@pytest.mark.unit
+def test_ballooning_stability_eval():
+    """Cross-compare all the stability functions.
+
+    We calculated the ideal ballooning growth rate and Newcomb ball
+    metric for the HELIOTRON case at different radii.
+    """
+    mu_0 = 4 * jnp.pi * 1e-7
```

Comment:
minor comment: usually we import `mu_0` from `scipy.constants`

Reply:
ok

Reply:
Ok, i'll do that

Reply:
Done.